iRODS 5 updates - configuration, process model, installation, genquery, etc. (main) (#338)
irods/irods_docs: docs/system_overview/genquery.md

GenQuery2 will notice the use of double single quotes and collapse them to one single quote before sending to the database. In the case of the example, that means `that''s` will be passed to the database as `that's`.

### Managing Duplicate/Unique Entries

> GenQuery2 no longer automatically inserts the DISTINCT keyword into the generated SQL. Users now have full control over the placement of the DISTINCT keyword.

Reviewer: ```suggestion
GenQuery2 does not automatically insert the DISTINCT keyword into the generated SQL. Users have full control over the placement of the DISTINCT keyword.
```

Author: Done.

!!! Important

>     The GenQuery2 implementation for iRODS 4.3.4 and earlier, originally removed duplicate entries from the resultset. This made it difficult for users to count records. Granting control over the DISTINCT keyword gives users more control and makes the behavior of the system more deterministic.

Author: Add a statement about using GenQuery2 with earlier versions of iRODS.

Author: Done.

>     The GenQuery2 implementation for iRODS 4.3.4 and earlier, originally removed duplicate entries from the resultset. This made it difficult for users to count records. Granting control over the DISTINCT keyword gives users more control and makes the behavior of the system more deterministic.

Reviewer: ```suggestion
    The initial GenQuery2 implementation for iRODS 4.3.4 and earlier removed duplicate entries from the resultset. This made it difficult for users to count records. Granting control over the DISTINCT keyword gives users more control and makes the behavior of the system more deterministic.
```

Author: Done.

> The following example now produces a resultset containing all replicas in the catalog.

Reviewer 2: Remove "now"?

Author: Done.
